examples/technical_blogs/tartan_blog/build_and_run_networks.py

```
@@ -0,0 +1,156 @@
+    try:
+        subprocess.run(cmake_command, check=True)
+        subprocess.run("ninja", check=True)
+    except subprocess.CalledProcessError as e:
+        print(f"Error executing command: {e}")
+
+# Build portBLAS
+def build_portBLAS():
+    script_directory = os.getcwd()
+    portBlas_build_dir_path = os.path.join(script_directory, "build_portBlas")
+    
+    create_directory(portBlas_build_dir_path)
+    os.chdir(portBlas_build_dir_path)
+    
+    cmake_portBlas_command = [
+        "cmake",
+        f"{portBLAS_source}",
+        "-GNinja",
+        "-DSYCL_COMPILER=dpcpp"
+    ]
+
+    run_cmake_and_ninja(cmake_portBlas_command)
+    os.chdir("..")
+
+# Build SYCL-DNN
+def build_sycl_dnn():
+    script_directory = os.getcwd()
+    sycl_dnn_build_dir = os.path.join(script_directory, "build_sycl_dnn")
+    
+    create_directory(sycl_dnn_build_dir)
+    os.chdir(sycl_dnn_build_dir)
+    
+    cmake_sycl_dnn_command = [
+        "cmake",
+        f"{SYCL_DNN_source}",
+        "-GNinja",
+        f"-DCMAKE_CXX_COMPILER={os.environ['CMAKE_CXX_COMPILER']}",
+        "-DSNN_BUILD_BENCHMARKS=OFF",
+        "-DSNN_BENCH_SYCLBLAS=OFF",
+        "-DSNN_BUILD_DOCUMENTATION=OFF",
+        f"-DSyclBLAS_DIR='{os.path.join(script_directory, 'build_portBlas')}'"
+    ]
+
+    run_cmake_and_ninja(cmake_sycl_dnn_command)
+    os.chdir("..")
+
+
+def download_and_convert_model(file_url, destination_file, h5_to_bin_script):
+    try:
+        response = requests.get(file_url)
+        response.raise_for_status()
+        with open(destination_file, "wb") as file:
+            file.write(response.content)
+        print(f"Downloaded '{destination_file}' successfully.")
+        
+        subprocess.run(["python", h5_to_bin_script, destination_file], check=True)
+        print(f"Converted '{destination_file}' to binary successfully.")
+    except requests.exceptions.RequestException as e:
+        print(f"Error downloading file: {e}")
```

Should the script terminate if it catches an exception in any of these files? Or should it try and continue on? Maybe we should re-throw an exception and catch it at the top-level? `try: main() except ...`?